README.md

```diff
@@ -200,132 +200,100 @@ Current Kubernetes Releases: https://kubernetes.io/releases/
+### Run the unit tests
 
-- **General Part**: Sets up the test env & checks if the HCCM Pod is properly running
-   - Build Tag: `e2e`
-- **Cloud Part**: Tests regular functionality against a Cloud-only environment
-   - Build Tag: `e2e && !robot`
-- **Robot Part**: Tests Robot functionality against a Cloud+Robot environment
-   - Build Tag: `e2e && robot`
+To run the unit tests, make sure you installed the following tools:
 
-## Local test setup
-This repository provides [skaffold](https://skaffold.dev/) to easily deploy / debug this controller on demand
+- [Go](https://go.dev/)
 
-### Requirements
-1. Install [hcloud-cli](https://github.com/hetznercloud/cli)
-2. Install [k3sup](https://github.com/alexellis/k3sup)
-3. Install [cilium](https://github.com/cilium/cilium-cli)
-4. Install [docker](https://www.docker.com/)
+1. Run the following command to run the unit tests:
 
-You will also need to set a `HCLOUD_TOKEN` in your shell session
-### Manual Installation guide
-1. Create an SSH key
-
-Assuming you already have created an ssh key via `ssh-keygen`
-```
-hcloud ssh-key create --name ssh-key-ccm-test --public-key-from-file ~/.ssh/id_rsa.pub 
+```sh
+go test ./...
 ```
 
-2. Create a server
-```
-hcloud server create --name ccm-test-server --image ubuntu-20.04 --ssh-key ssh-key-ccm-test --type cx22 
-```
+### Run the kubernetes e2e tests
 
-3. Setup k3s on this server
-```
-k3sup install --ip $(hcloud server ip ccm-test-server) --local-path=/tmp/kubeconfig --cluster --k3s-channel=v1.23 --k3s-extra-args='--no-flannel --no-deploy=servicelb --no-deploy=traefik --disable-cloud-controller --disable-network-policy --kubelet-arg=cloud-provider=external'
-```
-- The kubeconfig will be created under `/tmp/kubeconfig`
-- Kubernetes version can be configured via `--k3s-channel`
+Before running the e2e tests, make sure you followed the [Setup a development environment](#setup-a-development-environment) steps.
 
-4. Switch your kubeconfig to the test cluster. Very important: exporting this like 
-```
-export KUBECONFIG=/tmp/kubeconfig
-```
+1. Run the kubernetes e2e tests using the following command:
 
-5. Install cilium + test your cluster
-```
-cilium install
+```sh
+source dev/files/env.sh
+go test ./tests/e2e -tags e2e -v
 ```
 
-6. Add your secret to the cluster
-```
-kubectl -n kube-system create secret generic hcloud --from-literal="token=$HCLOUD_TOKEN"
-```
+### Development with Robot
 
-7. Deploy the hcloud-cloud-controller-manager
-```
-SKAFFOLD_DEFAULT_REPO=your_docker_hub_username skaffold dev
+If you want to work on the Robot support, you need to make some changes to the above setup.
+
+This requires that you have a Robot Server in the same account you use for the development. The server needs to be setup with the Ansible Playbook `dev/robot/install.yml` and configured in `dev/robot/install.yml`.
+
+1. Set these environment variables:
+
+```shell
+export TF_VAR_setup_robot=true
+export ROBOT_ENABLED=true
```

Comment:
Could we deduplicate this ? We should move the ROBOT_ENABLED var in the makefile, and write the env.autovars file during the setup.